lab-katy/route/image-router.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const del = require('del');
+const AWS = require('aws-sdk'); //software development kit
+const multer = require('multer');
+const Router = require('express').Router;
+const createError = require('http-errors');
+const debug = require('debug')('cfgram:image-router');
+
+const Image = require('../model/image.js');
+const Collection = require('../model/collection.js');
+const bearerAuth = require('../lib/bearer-auth-middleware.js');
+const imageRouter = module.exports = Router();
+
+AWS.config.setPromisesDependency(require('bluebird'));
+
+//instantiates new S3 model object that has tons of methods on it that allow things like upload/delete/ etc. Docs take some time to get through. look at upload and delete object method
+const s3 = new AWS.S3(); 
+const dataDir = `${__dirname}/../data`; //where data initially uploaded in file system
+const upload = multer({ dest: dataDir }); //tells multer where to go. (where our file is, and where to upload it) //https://stackoverflow.com/questions/4526273/what-does-enctype-multipart-form-data-mean
+
+function s3uploadProm(params) {
+  debug('s3uploadProm');
+
+  return new Promise((resolve) => {
```

Comment:
s3 actually has promises natively and you can access them by chaining `.promise()` on the end of the call.